### PR TITLE
oneDNN v3.6 release notes

### PR DESCRIPTION
**Pull request by vpirogov**
With growing amount of contributions to oneDNN capturing all the user-visible changes in human readable form became challenging. This PR includes a release notes draft based on the information from the PRs for the contributors to review. Your additions and corrections are highly appreciated.

### CODE REVIEW

**Review by tprimak — APPROVED** [review]

**Review by TaoLv — APPROVED** [review]

**Review by shu1chen — APPROVED** [review]

**Review by Shreyas-fuj — APPROVED** [review]

**Review by nikhilfujitsu — APPROVED** [review]

**Review by theComputeKid — APPROVED** [review]

**Review by mgouicem — APPROVED** [review]

**Review by ranukund — APPROVED** [review]
Minor changes suggested. Please incorporate as you see fit. Thanks!

**Review by kasturedeeksha — APPROVED** [review]

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+# Performance Optimizations
+
+## Intel Architecture Processors
+
+  * Improved performance for 4th generation Intel Xeon Scalable processors
+  (formerly Sapphire Rapids).
+  * Improved performance for Intel Xeon 6 processors (formerly Granite Rapids).
+  * Improved performance of group normalization primitive.
+  * Improved bf16 matmul performance with int4 compressed weights on processors
```
[the remark this answers is not in the document]
Comment (vpirogov) [reply RELEASE_NOTES.md:12]: This is a data type name, so it should not be capitalized. Added backticks to align with the rest of the document.

File: RELEASE_NOTES.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+# Usability
+
+  * Added [examples][Graph API examples] for SDPA, MQA, and GQA patterns
+  implementation with Graph API.
+  * Added [an example][deconvolution example] for deconvolution primitive.
+  * Added examples for [Vanilla RNN][Vanilla RNN example] and
+  [LBR GRU][LBR GRU example] RNN cells.
+  * Introduced support for Intel DPC++/C++ Compiler 2025.0.
+  * Introduced interoperability with [SYCL Graph] record/replay mode.
```
[the remark this answers is not in the document]
Comment (vpirogov) [reply RELEASE_NOTES.md:109]: The link works just fine. You can see rendered version of the document [in the branch](https://github.com/oneapi-src/oneDNN/blob/vpirogov/release-notes/RELEASE_NOTES.md)

File: RELEASE_NOTES.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+  * Removed dependency on OpenCL runtime for NVIDIA and AMD GPUs.
+  * **[experimental]** Introduced [logging mechanism][spdlog] based on spdlog
```
[the remark this answers is not in the document]
Comment (vpirogov) [reply RELEASE_NOTES.md:111]: We use [experimental] tag to communicate that certain features do not guarantee API stability and require explicit opt-in. If you have suggestions on how make it better please share.

**Comment by spalicki** [comment]
make test Lightweight_scans

File: RELEASE_NOTES.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+    Multi-Query Attention (MQA), and Grouped Query Attention (GQA) patterns.
+    - `LayerNorm`, `GroupNorm`, and `SoftMax` with `int8` quantized output
+    and zero-points.
+
+## Intel Graphics Products
+
+  * Improved performance for the Intel Data Center GPU Max Series (formerly
+  Ponte Vecchio).
+  * Introduced broad production quality optimizations for Intel Arc Graphics for
+  Intel Core Ultra Processors (Series 2) (formerly Lunar Lake).
```
Comment (mgouicem) [inline RELEASE_NOTES.md:33]:
```suggestion
  Intel Core Ultra processors (Series 2) (formerly Lunar Lake).
```

File: RELEASE_NOTES.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+  * Introduced broad production quality optimizations for future discrete GPU
+  based on Xe2 architecture (code name Battlemage).
+  * Introduced support for Intel Arc Graphics for future Intel Core Ultra
+  Processor (code name Arrow Lake-H).
```
Comment (mgouicem) [inline RELEASE_NOTES.md:37]:
```suggestion
  * Introduced support for Intel Arc Graphics for future 
  Intel Core Ultra processor (code name Arrow Lake-H).
```

File: RELEASE_NOTES.md [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+  * Improved performance of `fp8_e5m2` primitives on Intel Data Center GPU Max
+  Series (formerly Ponte Vecchio).
```
Comment (mgouicem) [inline RELEASE_NOTES.md:39]:
```suggestion
  * Improved performance of `fp8_e5m2` primitives on 
  Intel Data Center GPU Max Series (formerly Ponte Vecchio).
```

File: RELEASE_NOTES.md [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+  * Improved matmul and inner product primitives performance for shapes relevant
+  to large language models (LLMs) on GPUs with Intel XMX support.
+  * Improved `int8` convolution performance with weight zero-points.
+  * Reduced primitive creation time for softmax, layer normalization, and concat
+  primitives via kernel reuse.
+  * Improved performance of the following subgraphs with Graph API:
+    - SDPA without scale, MQA, and GQA patterns. `f16` variants of these
+    patterns significantly benefit from Intel(R) Xe Matrix Extensions (Intel(R)
+    XMX) support.
```
Comment (mgouicem) [inline RELEASE_NOTES.md:48]:
```suggestion
    patterns significantly benefit from Intel(R) Xe Matrix Extensions 
    (Intel(R) XMX) support.
```

File: RELEASE_NOTES.md [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+[sparse API]: https://oneapi-src.github.io/oneDNN/v3.6/dev_guide_experimental.html#onednn-experimental-sparse
+
+# Usability
+
+  * Added [examples][Graph API examples] for SDPA, MQA, and GQA patterns
+  implementation with Graph API.
+  * Added [an example][deconvolution example] for deconvolution primitive.
+  * Added examples for [Vanilla RNN][Vanilla RNN example] and
+  [LBR GRU][LBR GRU example] RNN cells.
+  * Introduced support for Intel DPC++/C++ Compiler 2025.0.
```
Comment (mgouicem) [inline RELEASE_NOTES.md:108]:
```suggestion
  * Introduced support for Intel oneAPI DPC++/C++ Compiler 2025.0.
```

File: RELEASE_NOTES.md [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+  to large language models (LLMs) on GPUs with Intel XMX support.
+  * Improved `int8` convolution performance with weight zero-points.
+  * Reduced primitive creation time for softmax, layer normalization, and concat
+  primitives via kernel reuse.
+  * Improved performance of the following subgraphs with Graph API:
+    - SDPA without scale, MQA, and GQA patterns. `f16` variants of these
+    patterns significantly benefit from Intel(R) Xe Matrix Extensions (Intel(R)
+    XMX) support.
+    - `fp8`, `convolution`, and `unary` or `binary` on the Intel Data Center GPU Max
+    Series.
```
Comment (mgouicem) [inline RELEASE_NOTES.md:50]:
```suggestion
    - `fp8`, `convolution`, and `unary` or `binary` on the 
    Intel Data Center GPU Max Series.
```

**Comment by mgouicem** [comment]
make test Lightweight_scans

**Comment by mgouicem** [comment]
make test Lightweight_scans

**Pull request merged**